### PR TITLE
testsuite: cannot case-switch on MPI handles

### PR DESCRIPTION
**Pull request by jeffhammond**
## Author Checklist
* [x] **Provide Description** 

It is a violation of the MPI standard to use switch-case on MPI handles, except for the ones that are explicitly stated to be compile-time constants.  See MPI-4 2.5.4 for details.

* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [ ] **Passes All Tests**

I will wait for CI.

* [?] **Contribution Agreement**

I assume I'm covered by the one Mellanox uses.

### CODE REVIEW

**Comment by hzhou** [comment]
Hi Jeff, thanks for the PR. Those are good fixes. Could you appease the spelling tsar and whitespace checker? They are a bit draconian to occasional contributors but do serve greater goods for regulars. As for the contribution agreement, we do have one for Mellanox, but I am not sure it can legally cover the team at present. Could you check with NVIDIA's legal and possibly get a new agreement signed. We'll need one with NVIDIA sooner or later anyway.

**Comment by hzhou** [comment]
@jeffhammond The way our CI checker works (per commit check), you have to squash the fixup commits in order for them to pass.

**Comment by hzhou** [comment]
Try use `maint/code-cleanup.bash [the_source_file] ` to clean up the white space.

**Comment by hzhou** [comment]
test:mpich/ch3/most
test:mpich/ch4/most

File: test/mpi/rma/rma_contig.c
Comment (jdinan) [inline test/mpi/rma/rma_contig.c:41]: The C type for these constants must be `const int (or unnamed enum)` (pg. 859 MPI 4.0). As such, they should be perfectly fine to use with `switch...case`.

Comment (dalcinl) [reply test/mpi/rma/rma_contig.c:41]: IIRC, `const int` is a compile time constant in C++, but not in C :disappointed: .

Comment (hzhou) [reply test/mpi/rma/rma_contig.c:41]: They are macro constants in MPICH - https://github.com/pmodels/mpich/blob/f51b6595223274eed72b3c6fcd5d8b49acd4b439/src/include/mpi.h.in#L396-L397

I wonder what was the issue 😕

Comment (dalcinl) [reply test/mpi/rma/rma_contig.c:41]: > I wonder what was the issue

Yes, although there is nothing wrong with the way MPICH does it. Users have to be careful. As the MPI standard says that "MPI constants" can be `const int`, portable MPI codes should not use them in `switch` statements. If the MPICH testsuite wants to be implementation-agnostic, then it should not use `switch statements with predefined/named handles and constants.

**Comment by dalcinl** [comment]
@jeffhammond Maybe you can rebase this PR to the latest main branch?

File: test/mpi/rma/rma_contig.c
Comment (hzhou) [reply test/mpi/rma/rma_contig.c:41]: Ah, thanks! I forgot to step outside MPICH.

Comment (jeffhammond) [reply test/mpi/rma/rma_contig.c:41]: Yes, my goal is to use the MPICH test suite for other implementations.

**Comment by hzhou** [comment]
@jeffhammond We are still waiting for the contribution agreement before we can merge the code. We will accept an individual agreement if your company is ok with it.

**Comment by jeffhammond** [comment]
I understand someone submitted the CLA recently. Please let me know if that means we can move forward.

**Review by hzhou — APPROVED** [review]
LGTM.

**Comment by hzhou** [comment]
test:mpich/authorship

**Pull request merged**
